internal/file/proto_set_provider.go

```diff
@@ -151,9 +162,15 @@ func (c *protoSetProvider) GetMultipleForFiles(workDirPath string, filePaths ...
 func (c *protoSetProvider) getBaseProtoSets(dirPathToProtoFiles map[string][]*ProtoFile) ([]*ProtoSet, error) {
 	filePathToProtoSet := make(map[string]*ProtoSet)
 	for dirPath, protoFiles := range dirPathToProtoFiles {
-		configFilePath, err := c.configProvider.GetFilePathForDir(dirPath)
-		if err != nil {
-			return nil, err
+		var configFilePath string
+		var err error
+		// we only want one ProtoSet if we have set configDirPath and configData
+		// since we are overriding all configuration files
+		if c.configDirPath == "" {
```

Comment:
We can update this to `c.configData == ""`

Reply:
Done